tests/src/test/scala/cats/tests/TryTests.scala

```diff
@@ -25,6 +24,9 @@ class TryTests extends CatsSuite {
   checkAll("Try", MonadTests[Try].monad[Int, Int, Int])
   checkAll("Monad[Try]", SerializableTests.serializable(Monad[Try]))
 
+  checkAll("Try[Int]", ApplicativeEvalTests[Try].applicativeEvalWithError[Int, Int, Int])
+  checkAll("ApplicativeEval[Try]", SerializableTests.serializable(ApplicativeEval[Try]))
```

Comment:
can we add a negative test to show that `ApplicativeEval[Try]` and `MonadDefer[Try]` will fail the laws?

I think Try should not be lawful if this typeclass is really representing anything. Since any `Applicative` can do it if `Try` can, no?